bench/clap-rs/optimized/4bajo035z6e1d4qz.ll

```diff
@@ -8983,111 +8983,109 @@ common.resume:                                    ; preds = %132, %80, %39
   %.not.i = xor i1 %106, true
   %109 = and i1 %107, %.not.i
   %110 = xor i1 %106, %107
-  br label %"_ZN71_$LT$core..option..Option$LT$T$GT$$u20$as$u20$core..cmp..PartialOrd$GT$11partial_cmp17hd2b066d5f3f3306aE.exit"
+  %..i = zext i1 %110 to i8
+  br i1 %109, label %"_ZN71_$LT$core..option..Option$LT$T$GT$$u20$as$u20$core..cmp..PartialOrd$GT$11partial_cmp17hd2b066d5f3f3306aE.exit.thread", label %"_ZN71_$LT$core..option..Option$LT$T$GT$$u20$as$u20$core..cmp..PartialOrd$GT$11partial_cmp17hd2b066d5f3f3306aE.exit"
 
 111:                                              ; preds = %101
-  %112 = icmp ult i8 %103, %105
-  %113 = icmp ne i8 %103, %105
+  %.0.i.i = call noundef range(i8 -1, 2) i8 @llvm.ucmp.i8.i8(i8 %103, i8 %105)
   br label %"_ZN71_$LT$core..option..Option$LT$T$GT$$u20$as$u20$core..cmp..PartialOrd$GT$11partial_cmp17hd2b066d5f3f3306aE.exit"
 
 "_ZN71_$LT$core..option..Option$LT$T$GT$$u20$as$u20$core..cmp..PartialOrd$GT$11partial_cmp17hd2b066d5f3f3306aE.exit": ; preds = %108, %111
-  %.sink3.i = phi i1 [ %110, %108 ], [ %113, %111 ]
-  %.sink.i = phi i1 [ %109, %108 ], [ %112, %111 ]
-  %not..sink.i = xor i1 %.sink.i, true
-  %114 = and i1 %.sink3.i, %not..sink.i
-  %spec.select = select i1 %114, ptr %98, ptr %26
-  br label %115
-
-115:                                              ; preds = %"_ZN71_$LT$core..option..Option$LT$T$GT$$u20$as$u20$core..cmp..PartialOrd$GT$11partial_cmp17hd2b066d5f3f3306aE.exit", %97
-  %.041 = phi ptr [ %26, %97 ], [ %spec.select, %"_ZN71_$LT$core..option..Option$LT$T$GT$$u20$as$u20$core..cmp..PartialOrd$GT$11partial_cmp17hd2b066d5f3f3306aE.exit" ]
+  %.0.i45 = phi i8 [ %.0.i.i, %111 ], [ %..i, %108 ]
+  %112 = icmp eq i8 %.0.i45, 1
```

Comment:
Not exactly a regression (previous IR looks worse?) but there is an optimization opportunity here: If we fold the icmp into the phi then the ucmp becomes an icmp and the other one folds to the zext argument.